Unable to run QuarkusTests in Eclipse since Quarkus 3.22

### Describe the bug

In Eclipse IDE, Right click on a file Run As-> Junit fails with the followiing;

```
java.lang.RuntimeException: Internal error. The test class class org.acme.GreetingResourceTest should have been loaded with a QuarkusClassLoader, but instead it was loaded with jdk.internal.loader.ClassLoaders$AppClassLoader@63c12fb0. This is caused by the FacadeClassLoader not correctly identifying this class as a QuarkusTest.
	at io.quarkus.test.junit.QuarkusTestExtension.getClassLoaderFromTestClass(QuarkusTestExtension.java:337)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:631)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:712)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
```

Runnin a single test from a `@QuarkuTest` works



### Expected behavior

Running a full file works

### Actual behavior

Exceptions happens

### How to Reproduce?

reproduder: code.quarkus.io and download the default project

Steps to reproduce
1 - Import the maven project into eclipse
2- right click on the test class in the package explorer, or on the class name within the editor
3- Run As-> Junit Test


### Output of `uname -a` or `ver`

Darwin Kernel Version 24.4.0

### Output of `java -version`

Temurin-21.0.7+6

### Quarkus version or git rev

22.0.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

mvn 3.9.9

### Additional information

Running a single test seem to work ok.

I get the error when running tests during `./gradlew quarkusDev`

```
<====2025-05-07 15:24:37,632 INFO  [io.qua.test] (Quarkus Terminal Reader) Test output enabled
Starting tests
Press [e] to edit command line args (currently ''), [r] to re-run, [h] for more options>
Press [e] to edit command line args (currently ''), [r] to re-run, [o] Toggle test output, [h] for more options>
Running tests for the first time1s]
Press [e] to edit command line args (currently ''), [h] for more options>
Press [e] to edit command line args (currently ''), [o] Toggle test output, [h] for more options>

2025-05-07 15:24:38,911 ERROR [io.qua.test] (Test runner thread) Internal error running tests: java.lang.RuntimeException: java.util.NoSuchElementException: No value present
        at io.quarkus.deployment.dev.testing.JunitTestRunner.prepare(JunitTestRunner.java:447)
        at io.quarkus.deployment.dev.testing.ModuleTestRunner.prepare(ModuleTestRunner.java:83)
        at io.quarkus.deployment.dev.testing.TestSupport.runInternal(TestSupport.java:430)
        at io.quarkus.deployment.dev.testing.TestSupport$3.run(TestSupport.java:389)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.util.NoSuchElementException: No value present
        at java.base/java.util.Optional.get(Optional.java:143)
        at io.quarkus.deployment.dev.testing.JunitTestRunner.discoverTestClasses(JunitTestRunner.java:716)
        at io.quarkus.deployment.dev.testing.JunitTestRunner.prepare(JunitTestRunner.java:166)
        ... 4 more

```

--- a/test-framework/junit5/src/main/java/io/quarkus/test/junit/AbstractJvmQuarkusTestExtension.java
+++ b/test-framework/junit5/src/main/java/io/quarkus/test/junit/AbstractJvmQuarkusTestExtension.java
@@ -160,9 +160,20 @@ public class AbstractJvmQuarkusTestExtension extends AbstractQuarkusTestWithCont
                     .unwrap(SmallRyeConfig.class)
                     .getConfigMapping(TestConfig.class);
         } catch (Exception | ServiceConfigurationError e) {
-            // Tracked by https://github.com/quarkusio/quarkus/issues/46048
-            Log.error("Could not read configuration while evaluating whether to run " + context.getRequiredTestClass()
-                    + ". This usually happens when re-running a test that has already failed, for example if surefire.rerunFailingTestsCount is set. To work around this limitation, either adjust the test so that it passes, or isolate the test into a project whose tests all use the same combination of @TestProfile and resources.");
+            boolean isEclipse = System.getProperty("sun.java.command") != null
+                    && System.getProperty("sun.java.command").contains("JUnit5TestLoader");
+            if (isEclipse) {
+                // Tracked by https://github.com/eclipse-jdt/eclipse.jdt.ui/issues/2257
+                Log.error(
+                        "Could not read configuration while evaluating whether to run a test. This is a known issue when running tests in the Eclipse IDE. To work around the problem, edit the run configuration and add `-uniqueId [engine:junit-jupiter]/[class:"
+                                + context.getRequiredTestClass().getName()
+                                + "]` in the program arguments. Running the whole package, or running individual test methods, will also work without any extra configuration.");
+            } else {
+                Log.error("Internal error: Could not read configuration while evaluating whether to run "
+                        + context.getRequiredTestClass()
+                        + ". Please let the Quarkus team know what you were doing when this error happened.");
+
+            }
             Log.debug("Underlying exception: " + e);
             Log.debug("Thread Context Classloader: " + Thread.currentThread().getContextClassLoader());
             Log.debug("The class of the class we use for mapping is " + TestConfig.class.getClassLoader());

